CsharpExtrasTest/Extensions/ArrayExtension2DTest.cs

```diff
@@ -388,6 +399,14 @@ private static IEnumerable<object[]> ProviderForFoldingArrayToSingleColumn()
                 new object[] { new string[,] { { "a" }, { "1"} }, (Func<string, string, string>)((a, b) => a + b), new string[] { "a", "1" }, },
             };
         }
+
+        private static IEnumerable<object[]> ProviderForArrayToJaggedArray()
+        {
+            return new List<object[]> {
+                new object[] { new int[,] { { 1,2},{3,4 } } ,new int[][] { new int[] { 1, 2 }, new int[] { 3, 4 } }},
+                new object[] { new int[,] { { -1, -2 }, { -3, -4 } }, new int[][] { new int[] { -1, -2 }, new int[] { -3, -4 } } }

```

Comment:
This test case isn't massively different to the first: given that the method is generic it can't know what the objects look like so structurally it just sees black boxes, so in this sense this test case is more or less the same as the other one.

What would be better would be qualitatively different test cases e.g. what happens for an empty array? A single row? A single column?